File: 2095-delete-the-middle-node-of-a-linked-list/2095-delete-the-middle-node-of-a-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def getLength(self, head):
        tmp = head
        n = 0
        
        while tmp != None:
            tmp = tmp.next
            n += 1
            
        return n
        
    def deleteMiddle(self, head: Optional[ListNode]) -> Optional[ListNode]:
        
        if head.next == None:
            return None
        
        n = self.getLength(head)
        mid = n // 2
        i = 0
        tmp = head
        
        while i < mid - 1:
            tmp = tmp.next
            i += 1
        
        tmp.next = tmp.next.next
        
        return head
        

File: 2095-delete-the-middle-node-of-a-linked-list/2095-delete-the-middle-node-of-a-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* deleteMiddle(ListNode* head) {
        if(head -> next == nullptr) {
            head = nullptr;
            return head;
        }
        
        int n = getLength(head);
        int mid = n / 2;
        int i = 0;
        auto tmp = head;
        
        while(i < mid - 1) {
            tmp = tmp -> next;
            ++i;
        }
        
        tmp -> next = tmp -> next -> next;
        return head;
    }
private:
    int getLength(ListNode* head) {
        auto tmp = head;
        int n = 0;
        
        while(tmp != nullptr) {
            tmp = tmp -> next;
            ++n;
        }
        return n;
    }
};